sdks/python/apache_beam/metrics/monitoring_infos.py

```diff
@@ -0,0 +1,246 @@
+from apache_beam.metrics.cells import GaugeData
+from apache_beam.metrics.cells import GaugeResult
+from apache_beam.portability import common_urns
+from apache_beam.portability.api.beam_fn_api_pb2 import CounterData
+from apache_beam.portability.api.beam_fn_api_pb2 import Metric
+from apache_beam.portability.api.beam_fn_api_pb2 import MonitoringInfo
+
+ELEMENT_COUNT_URN = common_urns.monitoring_infos.ELEMENT_COUNT.urn
+START_BUNDLE_MSECS_URN = common_urns.monitoring_infos.START_BUNDLE_MSECS.urn
+PROCESS_BUNDLE_MSECS_URN = common_urns.monitoring_infos.PROCESS_BUNDLE_MSECS.urn
+FINISH_BUNDLE_MSECS_URN = common_urns.monitoring_infos.FINISH_BUNDLE_MSECS.urn
+TOTAL_MSECS_URN = common_urns.monitoring_infos.TOTAL_MSECS.urn
+USER_COUNTER_URN_PREFIX = (
+    common_urns.monitoring_infos.USER_COUNTER_URN_PREFIX.urn)
+
+# TODO(ajamato): Implement the remaining types, i.e. Double types
+# Extrema types, etc. See:
+# https://s.apache.org/beam-fn-api-metrics
+SUM_INT64_TYPE = common_urns.monitoring_info_types.SUM_INT64_TYPE.urn
+DISTRIBUTION_INT64_TYPE = (
+    common_urns.monitoring_info_types.DISTRIBUTION_INT64_TYPE.urn)
+LATEST_INT64_TYPE = common_urns.monitoring_info_types.LATEST_INT64_TYPE.urn
+
+COUNTER_TYPES = set([SUM_INT64_TYPE])
+DISTRIBUTION_TYPES = set([DISTRIBUTION_INT64_TYPE])
+GAUGE_TYPES = set([LATEST_INT64_TYPE])
+
+
+def to_timestamp_proto(timestamp_secs):
+  """Converts seconds since epoch to a google.protobuf.Timestamp.
+
+  Args:
+    timestamp_secs: The timestamp in seconds since epoch.
+  """
+  seconds = int(timestamp_secs)
+  nanos = int((timestamp_secs - seconds) * 10**9)
+  return timestamp_pb2.Timestamp(seconds=seconds, nanos=nanos)
+
+
+def to_timestamp_secs(timestamp_proto):
+  """Converts a google.protobuf.Timestamp to seconds since epoch.
+
+  Args:
+    timestamp_proto: The google.protobuf.Timestamp.
+  """
+  return timestamp_proto.seconds + timestamp_proto.nanos * 10**-9
+
+
+def extract_counter_value(monitoring_info_proto):
+  """Returns the int coutner value of the monitoring info."""
+  if is_counter(monitoring_info_proto) or is_gauge(monitoring_info_proto):
+    return monitoring_info_proto.metric.counter_data.int64_value
+  return None
+
+
+def extract_distribution(monitoring_info_proto):
+  """Returns the relevant DistributionInt64 or DistributionDouble.
+
+  Args:
+    monitoring_info_proto: The monitoring infor for the distribution.
+  """
+  if is_distribution(monitoring_info_proto):
+    return monitoring_info_proto.metric.distribution_data.int_distribution_data
+  return None
+
+
+def create_labels(ptransform='', tag=''):
+  """Create the label dictionary based on the provided tags.
+
+  Args:
+    ptransform: The ptransform/step name.
+    tag: he output tag name, used as a label.
+  """
+  labels = {}
+  if tag:
+    labels['TAG'] = tag
+  if ptransform:
+    labels['PTRANSFORM'] = ptransform
+  return labels
+
+
+def int64_counter(urn, metric, ptransform='', tag=''):
```

Comment:
This code feels very repetitive; isn't this excactly the kind of thing we should be pulling out of the yaml file?